src/background/services/paymentSession.ts

```diff
@@ -122,6 +127,12 @@ export class PaymentSession {
 
     let outgoingPayment: OutgoingPayment | undefined
 
+    const waitTime = await this.tabState.getOverpayingWaitTime(this.tab, this.url, this.receiver)
+
+    if (waitTime) {
+      await sleep(waitTime)
+    }
+
```

Comment:
Nit: If we always return `number` (0 in other case), we just `await sleep(0)`. Less things to think of (less branches).

Reply:
from what I know sleep 0 will still put the promise in the queue and it's picked up immediately after. I will keep it how we have it

Reply:
Which is fine. Very little impact on performance. It'll get optimized by browser.
I prefer to make code simpler to readers - so less decisions in mind, the better. A nitpick anyway, feel free to ignore